### PR TITLE
Use same arguments for gcc and clang.

### PR DESCRIPTION
**Pull request by mpoeter**
### Scope & Purpose

Fix some inconsistencies in the arguments passed to the compiler when building with clang.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.

Enterprise PR: https://github.com/arangodb/enterprise/pull/592

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12819/

### CODE REVIEW

File: arangod/CMakeLists.txt [hunk 1]
```diff
@@ -839,7 +839,7 @@ add_library(arango_rocksdb STATIC
   ${ADDITIONAL_LIB_ARANGO_ROCKSDB_SOURCES}
 )
 
-if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_CLANG)
   target_compile_options(arango_rocksdb PUBLIC "-maes")
```
Comment (jsteemann) [inline arangod/CMakeLists.txt:843]: I think this change is good. 
After it is merged, we can check whether we can roll back part of the changes in enterprise/c7dea6bfba5eea5c1bad581e492e721a7184bd91, which disabled AES intrinsics on MacOS.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -294,11 +294,11 @@ target_include_directories(arangodbtests PRIVATE
   ${INCLUDE_DIRECTORIES}
 )
 
-if(MSVC AND FAIL_ON_WARNINGS)
+if(MSVC AND USE_FAIL_ON_WARNINGS)
   target_compile_options(arangodbtests PRIVATE /WX /D_WINSOCK_DEPRECATED_NO_WARNINGS)
 endif()
 
-if (DARWIN AND FAIL_ON_WARNINGS)
+if (DARWIN AND USE_FAIL_ON_WARNINGS)
```
[the remark this answers is not in the document]
Comment (dothebart) [reply tests/CMakeLists.txt:301]: most probably some of them have to be fixedFirst?

**Review by dothebart — APPROVED** [review]

**Comment by mpoeter** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12853/

**Comment by mpoeter** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12854/

**Comment by mpoeter** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12864/

**Pull request merged**